opm/simulators/wells/ALQState.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+  Copyright 2021 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <cstddef>
+#include <stdexcept>
+
+
+#include <opm/simulators/wells/ALQState.hpp>
+
+namespace Opm {
+
+double ALQState::get(const std::string& wname) const {
+    auto iter = this->current_alq_.find(wname);
+    if (iter != this->current_alq_.end())
+        return iter->second;
+
+    auto default_iter = this->default_alq_.find(wname);
+    if (default_iter != this->default_alq_.end())
+        return default_iter->second;
+
+    throw std::logic_error("No ALQ value registered for well: " + wname);
+}
+
+void ALQState::update_default(const std::string& wname, double value) {
+    auto default_iter = this->default_alq_.find(wname);
+    if (default_iter == this->default_alq_.end() || default_iter->second != value) {
+        this->default_alq_[wname] = value;
+        this->current_alq_[wname] = value;
```

Comment:
I didn't notice this before, but if we want to be ever so slightly faster here, we can use the [`insert_or_assign()`](https://en.cppreference.com/w/cpp/container/map/insert_or_assign) member function which is new in C++17:
```C++
this->default_alq.insert_or_assign(wname, value);
this->default_alq.insert_or_assign(wname, value);
```